airflow/providers/amazon/aws/operators/rds.py

```diff
@@ -0,0 +1,605 @@
+        - status is in `wait_statuses`
+        - status not in `ok_statuses` and `error_statuses`
+        - `_describe_item()` returns non-empty list
+        """
+        while True:
+            items = self._describe_item(**kwargs)
+
+            if len(items) == 0:
+                break
+            elif len(items) > 1:
+                raise AirflowException(f"There is more than one item with the same identifier: {items}")
+
+            if wait_statuses and items[0]['Status'] in wait_statuses:
+                continue
+            elif ok_statuses and items[0]['Status'] in ok_statuses:
+                break
+            elif error_statuses and items[0]['Status'] in error_statuses:
+                raise AirflowException(f"All items have error statuses: {items}")
+
+            time.sleep(self._wait_interval)
+
+        return items
+
+    def execute(self, context: 'Context') -> str:
+        """Different implementations for snapshots, tasks and events"""
+        raise NotImplementedError
+
+    def on_kill(self) -> None:
+        """Different implementations for snapshots, tasks and events"""
+        raise NotImplementedError
+
+
+class RdsCreateDbSnapshotOperator(BaseRdsOperator):
+    """
+    Creates a snapshot of a DB instance or DB cluster.
+    The source DB instance or cluster must be in the available or storage-optimization state.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:RdsCreateDbSnapshotOperator`
+
+    :param db_type: Type of the DB - either "instance" or "cluster"
+    :type db_type: RDSDbType
+    :param db_identifier: The identifier of the instance or cluster that you want to create the snapshot of
+    :type db_identifier: str
+    :param db_snapshot_identifier: The identifier for the DB snapshot
+    :type db_snapshot_identifier: str
+    :param tags: A list of tags in format `[{"Key": "something", "Value": "something"},]
+        `USER Tagging <https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Tagging.html>`__
+    :type tags: Sequence[TagTypeDef] or None
+    """
+
+    template_fields = ("db_snapshot_identifier", "db_instance_identifier", "tags")
+
+    def __init__(
+        self,
+        *,
+        db_type: str,
+        db_identifier: str,
+        db_snapshot_identifier: str,
+        tags: Optional[Sequence[TagTypeDef]] = None,
+        aws_conn_id: str = "aws_conn_id",
+        **kwargs,
+    ):
+        super().__init__(aws_conn_id=aws_conn_id, **kwargs)
+        self.db_type = RdsDbType(db_type)
+        self.db_identifier = db_identifier
+        self.db_snapshot_identifier = db_snapshot_identifier
+        self.tags = tags or []
+
+    def _describe_item(self, **kwargs) -> list:
+        """Returns snapshot info"""
+        if self.db_type.value == "instance":
+            db_snapshots = self.hook.conn.describe_db_snapshots(
```

Comment:
now that you are in the operator you are freed to create a `client` cached property which seems a little cleaner / more intuitive so that you can do `self.client.describe...`

Reply:
Yeap. Thanks for that.
I see that many operators are using `hook.get_conn().method...`. 
Maybe later I could refactor these usages?

Reply:
Yep. Can be done later.